src/main/java/com/google/sps/data/SaleCard.java

```diff
@@ -0,0 +1,125 @@
+import java.time.LocalTime;
+
+/** A saleCard. */
+public final class SaleCard {
+
+  private final long id;
+  private String businessName;
+  private String description;
+  private boolean hasDelivery;
+  private LocalTime startTime;
+  private LocalTime endTime;
+  private LocationData location;
+  private Picture picture;
+
+  public SaleCard(long id, String businessName, String description, boolean hasDelivery,
+      LocalTime startTime, LocalTime endTime, LocationData location, Picture picture) {
+    this.id = id;
+    this.businessName = businessName;
+    this.description = description;
+    this.hasDelivery = hasDelivery;
+    this.startTime = startTime;
+    this.endTime = endTime;
+    this.location = location;
+    this.picture = picture;
+  }
+
+  public SaleCard(EmbeddedEntity embeddedSaleCard) {
+    if (embeddedSaleCard == null) {
+      this.id = -1;
+      return;
+    }
+
+    this.id = (long) embeddedSaleCard.getKey().getId();
+    this.businessName = (String) embeddedSaleCard.getProperty("businessName");
+    this.description = (String) embeddedSaleCard.getProperty("description");
+    this.hasDelivery = (boolean) embeddedSaleCard.getProperty("hasDelivery");
+
+    CharSequence start = (CharSequence) embeddedSaleCard.getProperty("startTime");
+    this.startTime = LocalTime.parse(start);
+    CharSequence end = (CharSequence) embeddedSaleCard.getProperty("endTime");
+    this.endTime = LocalTime.parse(end);
+    
+    EmbeddedEntity embeddedLocation = (EmbeddedEntity) embeddedSaleCard.getProperty("location");
+    this.location = (embeddedLocation == null) ? null : new LocationData(embeddedLocation);
+    EmbeddedEntity embeddedPicture = (EmbeddedEntity) embeddedSaleCard.getProperty("picture");
+    this.picture = (embeddedPicture == null) ? null : new Picture(embeddedPicture);
+  }
+
+  public long getId() {
+    return id;
+  }
+
+  public String getBusinessName() {
+    return businessName;
+  }
+
+  public String getDescription() {
+    return description;
+  }
+
+  public boolean hasDelivery() {
+    return hasDelivery;
+  }
+
+  public LocalTime getStartTime() {
+    return startTime;
+  }
+
+  public LocalTime getEndTime() {
+    return endTime;
+  }
+
+  public LocationData getLocation() {
+    return location;
+  }
+
+  public Picture getPic() {
+    return picture;
+  }
+
+  public void setBusinessName(String businessName) {
+    this.businessName = businessName;
+  }
+
+  public void setDescription(String description) {
+    this.description = description;
+  }
+
+  public void setHasDelivery(boolean hasDelivery) {
+    this.hasDelivery = hasDelivery;
+  }
+
+  public void setStartTime(LocalTime startTime) {
+    this.startTime = startTime;
+  }
+
+  public void setEndTime(LocalTime endTime) {
+    this.endTime = endTime;
+  }
+
+  public void setLocation(LocationData location) {
+    this.location = location;
+  }
+
+  public void setPic(Picture picture) {
```

Comment:
Can we change this to a more descriptive name?
```suggestion
  public void setPicture(Picture picture) {
```

Reply:
Sure! Thanks for noticing it!